### PR TITLE
New serverless pattern - SQS-EBPipes-ECS-Task Typescript CDK

### PR DESCRIPTION
**Pull request by virajpadte**
*Description of changes:*
Adding a new pattern for triggering ECS Fargate tasks from matches SQS events via EventBridge Pipes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/lib/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    // Give pipeRole permissions to consume from the SQS queue and run the ECS task
+    sqsQueue.grantConsumeMessages(pipeRole);
+    // sqsEventBridgePipeTargetLogGroup.grantWrite(pipeRole);
+    testfargateTaskDefinition.grantRun(pipeRole);
+    eventBridgePipeTraceLogGroup.grantWrite(pipeRole);
+
+    // create a EventBridge pipe which consumes events from SQS with batch size 1 and delivers events to CloudWatch log group sqsEventBridgePipeTargetLogGroup
+    const pipe = new pipes.CfnPipe(this, 'SQSEventBridgePipe', {
+      logConfiguration: {
+        level: 'TRACE',
```
[the remark this answers is not in the document]
Comment (virajpadte) [reply eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/lib/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack.ts:93]: No I had this in there during debugging I will clean this up and clean it all up.

File: eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/jest.config.js
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/jest.config.js:1]: I think this file can be removed as you have no tests

File: eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/lib/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+
+    const sqsDLQ = new Queue(this, 'SQSDLQ', {
+      queueName: 'sqs-dlq',
+    });
+
+    // create a SQS queue
+    const sqsQueue = new sqs.Queue(this, 'SQSQueue', {
+      queueName: 'sqs-queue',
+      visibilityTimeout: cdk.Duration.seconds(300),
```
[the remark this answers is not in the document]
Comment (virajpadte) [reply eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/lib/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack.ts:25]: no real reason - removed

File: eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/lib/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+export class EventbridgePipesSqsToEcsTaskCdkTypescriptStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+
+    const sqsDLQ = new Queue(this, 'SQSDLQ', {
+      queueName: 'sqs-dlq',
+    });
+
+    // create a SQS queue
```
[the remark this answers is not in the document]
Comment (virajpadte) [reply eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/lib/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack.ts:22]: valid - removed!

File: eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/lib/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+
+    // create a IAM role for EventBridge pipe
+    const pipeRole = new iam.Role(this, 'PipeRole', {
+      assumedBy: new iam.ServicePrincipal('pipes.amazonaws.com'),
+    });
+
+    // Give pipeRole permissions to consume from the SQS queue and run the ECS task
+    sqsQueue.grantConsumeMessages(pipeRole);
+    // sqsEventBridgePipeTargetLogGroup.grantWrite(pipeRole);
```
[the remark this answers is not in the document]
Comment (virajpadte) [reply eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/lib/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack.ts]: removed!

File: eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/test/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript.test.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+// import * as cdk from 'aws-cdk-lib';
+// import { Template } from 'aws-cdk-lib/assertions';
+// import * as EventbridgePipesSqsToEcsTaskCdkTypescript from '../lib/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack';
+
+// example test. To run these tests, uncomment this file along with the
+// example resource in lib/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack.ts
+test('SQS Queue Created', () => {
```
[the remark this answers is not in the document]
Comment (virajpadte) [reply eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/test/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript.test.ts]: no removed the folder

File: eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/example-pattern.json [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        "text": [
+            "This pattern demonstrates how to invoke an Amazon Elastic Container Service Tasks based on selective messages from Amazon SQS queue using Amazon EventBridge Pipes."
+        ]
+    },
+    "gitHub": {
+        "template": {
+            "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript",
+            "templateURL": "serverless-patterns/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript",
+            "projectFolder": "eventbridge-pipes-sqs-to-ecs-task-cdk-typescript",
+            "templateFile": "lib/eventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack.ts"
```
[the remark this answers is not in the document]
Comment (virajpadte) [reply eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/example-pattern.json:18]: @bfreiberg I am not sure if I understand this. I changed the name to bin/app.ts and also changed in the file cdk.json. 
What exact change is needed in this file... do I need the lib/ventbridge-pipes-sqs-to-ecs-task-cdk-typescript-stack.ts to also be renamed to lib/app.ts or lib/stack.ts ?

File: eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/jest.config.js
Comment (virajpadte) [reply eventbridge-pipes-sqs-to-ecs-task-cdk-typescript/jest.config.js:1]: removed!

**Pull request merged**